lib/events.js

```diff
@@ -353,8 +374,21 @@ class AdobeIOEvents {
             body: JSON.stringify(body),
             retryOptions: retryOptions
         });
-
-        return handleFetchErrors(response).json();
+        try {
+            response = handleFetchErrors(response).json();
+            return response;
+        }
+        catch (e) {
+            if (e.statusCode === 409) {
+                console.error("Journal already exists, returning existing");
+                const headers = response.headers.raw();
+                const conflictingRegistrationId = headers['x-conflicting-id'][0];
+                if (!headers || !conflictingRegistrationId) {
+                    throw e;
+                }
+                return this.getJournal(journal, conflictingRegistrationId);
+            }
```

Comment:
This is swallowing exceptions for errors where the failure was not not a 409 failure response. It should be re-throwing exceptions that it can't handle.

Reply:
+1 
how about checking for `409` before `response = handleFetchErrors(response).json();`, instead of the catch()